### PR TITLE
Todo

### PR DESCRIPTION
**Pull request by BogdanIrimie**
The todo app has moved to https://github.com/aserto-demo/todo-java-v2
Removed the part that reads from `.env` filed from the config.
Add a DirectoryClient, this should be extended in the future.

### CODE REVIEW

**Review by apoland — APPROVED** [review]

File: src/main/java/com/aserto/AuthzClient.java [hunk 1]
```diff
@@ -17,10 +17,9 @@
 
 public class AuthzClient implements AuthorizerClient {
     private AuthorizerGrpc.AuthorizerBlockingStub client;
-    private ManagedChannel channel;
     public AuthzClient(ManagedChannel channel) {
         client = AuthorizerGrpc.newBlockingStub(channel);
-        this.channel = channel;
+        AuthorizerGrpc.newStub(channel);
```
[the remark this answers is not in the document]
Comment (BogdanIrimie) [reply src/main/java/com/aserto/AuthzClient.java:22]: This was just an experiment I did and forgot to remove the client. I will remove it.
Just for some context, the `AuthorizerGrpc.newStub(...)` creates an async client. Right now the Java SDK only supports sync calls, but we can add async calls as well in a future PR. Created an issue so we don't forget about it. https://github.com/aserto-dev/workspace/issues/147

File: src/main/java/com/aserto/DirectoryClient.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    public ImporterGrpc.ImporterBlockingStub getImporterClient() {
+        return importerClient;
+    }
+
+    public ExporterGrpc.ExporterBlockingStub getExporterClient() {
+        return exporterClient;
+    }
+
+    public com.aserto.directory.common.v2.Object[] getObjects() {
```
[the remark this answers is not in the document]
Comment (BogdanIrimie) [reply src/main/java/com/aserto/DirectoryClient.java]: Added parameters for type, page size and next token

File: src/test/java/AuthzClientIntegrationTest.java [hunk 3]
```diff
@@ -29,7 +30,7 @@ void testBuildAuthzClient() throws IOException {
 
         // Act
         List<Module> policies = authzClient.listPolicies("todo", "todo");
-        authzClient.close();
+        channel.shutdown();
```
[the remark this answers is not in the document]
Comment (BogdanIrimie) [reply src/test/java/AuthzClientIntegrationTest.java:33]: Initially, I did exactly that but later thought that the one that creates the object should handle its lifecycle. I am ok with both versions, so if there is a preference for one I can do the change.

[the remark this answers is not in the document]
Comment (ronenh) [reply src/test/java/AuthzClientIntegrationTest.java:33]: I think that closing the channel from the client gives users the same amount of control without requiring them to manage two separate objects.
If a user wants to use a single channel with multiple stubs, they can simply not call close on the authz client and only close the channel when they're done with it.

On the other hand, if a user just wants an authorizer client and doesn't care about reusing the underlying channel, they can call close on the client when they're done with it.

The client doesn't have any resources of its own besides the channel.

[the remark this answers is not in the document]
Comment (BogdanIrimie) [reply src/test/java/AuthzClientIntegrationTest.java]: Added back the close method

**Review by apoland — APPROVED** [review]

**Pull request merged**